ivadomed/training.py

```diff
@@ -54,6 +55,12 @@ def train(model_params, dataset_train, dataset_val, training_params, path_output
     # Write the metrics, images, etc to TensorBoard format
     writer = SummaryWriter(log_dir=path_output)
 
+    # Collect all hyperparameters into a dictionary
+    cfg = { **training_params, **model_params}
+    # Initialize WandB with metrics and hyperparameters
+    wandb.init(project=wandb_params["project_name"], group=wandb_params["group_name"], 
+                                name=wandb_params["run_name"], config=cfg)
```

Comment:
Again, we should not always assume the `wand_params` parental key even exists here. Here using the` .get()`  may still cause a crash as the `wand_params` could be `None` and cannot call `.get()` on `None`.  For more about [why use get](https://towardsdatascience.com/18-common-python-anti-patterns-i-wish-i-had-known-before-44d983805f0f#4d61) and [SO simple explanations](https://stackoverflow.com/a/11041421).

I recommend something like this: 
```
# Establish "wandb_params" exists and also its has a API key in it. 
if wandb_params and wandb_params[WandbKW.API_KEY]:

  # Setup default names for wandb.init, not assuming they exists by providing default alternatives.
   project_name = wand_params.get("WandbKW.PROJECT_NAME", "Default_Project")
   group_name= wand_params.get("WandbKW.GROUP_NAME", "Default_Group")
   run_name = wand_params.get("WandbKW.RUN_NAME", "Default_Run")
   
   # Login OR you can also use os.environ["WANDB_API_KEY"] to set the key into the environment. 
   wandb.login(key=wandb_params[WandbKW.API_KEY])

   # Init Project
   wandb.init(project=project_name, group=group_name , name=run_name, config=cfg)
```

Reply:
This suggestion and the reference links were super useful! Thank you very much for that! I do think having `wandb.login(...)` right here would be useful in terms of code-understandability (as important wandb methods would be at one place), instead of setting the environment variable. lmk if we should proceed with `os.environ` in case it has any other advantages!

Reply:
👍 for `wandb.login`. 

Since you asked, I think `os.environ` approach's only other pro I can potentially think of is when you set it, you can potentially not need to change it again (as API keys tend to stick around). So in "theory" if you say update `.bashrc` to set this or something similar semi-permanent, it will always prevent you ever worry about it. Easy from a seasoned ivadomed dev/user/experimenter perspective but not sure how easy that it would be for not experienced Linux users etc. Alternative to that would be load config and set the os environment but it usually doesn't stick around permanently so probably not worth the hassle. 

All in short, I agree `wandb.login` call is probably far more scalable and readable.